### DISCUSSION TITLE
Difference Between MicroPython ESP8266 and ESP32?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16371] [General]
In ESP32 MP I can do...
```
def Test():
    pass
print(Test.__name__)
```
In ESP8266 I get...
```
def Test():
    pass
print(Test.__name__)
AttributeError: 'function' object has no attribute '__name__'
```
This difference by design or intent?  Working on code that is pretty much generic across ESP32 and ESP8266, there are some differences to be sure, but this one just seemed odd.

**Comment by andrewleech** [comment]
The 8266 is a much smaller part in terms of flash and ram. As such by default it has a number of internal features disabled to save on resource usage.

**Comment by Jibun-no-Kage** [comment]
Cool  that would explain it.  Thanks.
